pkg/controller.v1beta1/trial/trial_controller_test.go

```diff
@@ -290,14 +287,55 @@ func TestReconcileBatchJob(t *testing.T) {
 		}, timeout).Should(gomega.BeTrue())
 	})
 
-	t.Run(`Trail with "Complete" BatchJob and Unavailable metrics.`, func(t *testing.T) {
+	t.Run(`Trial with "Complete" BatchJob and Unavailable metrics(StdOut MC).`, func(t *testing.T) {
 		g := gomega.NewGomegaWithT(t)
 		gomock.InOrder(
 			mockManagerClient.EXPECT().GetTrialObservationLog(gomock.Any()).Return(observationLogUnavailable, nil).MinTimes(1),
 			mockManagerClient.EXPECT().DeleteTrialObservationLog(gomock.Any()).Return(nil, nil),
 		)
-		// Create the Trial
-		trial := newFakeTrialBatchJob()
+		// Create the Trial with StdOut MC
+		trial := newFakeTrialBatchJob(commonv1beta1.StdOutCollector, "test-unavailable-stdout")
+		trialKey := types.NamespacedName{Name: "test-unavailable-stdout", Namespace: namespace}
+		g.Expect(c.Create(ctx, trial)).NotTo(gomega.HaveOccurred())
+
+		// Expect that Trial status is succeeded with "false" status and "metrics unavailable" reason.
+		// Metrics unavailable because GetTrialObservationLog returns "unavailable".
+		g.Eventually(func() bool {
+			if err = c.Get(ctx, trialKey, trial); err != nil {
+				return false
+			}
+			return trial.IsMetricsUnavailable() &&
+				len(trial.Status.Observation.Metrics) > 0 &&
+				trial.Status.Observation.Metrics[0].Min == consts.UnavailableMetricValue &&
+				trial.Status.Observation.Metrics[0].Max == consts.UnavailableMetricValue &&
+				trial.Status.Observation.Metrics[0].Latest == consts.UnavailableMetricValue
+		}, timeout).Should(gomega.BeTrue())
```

Comment:
```suggestion
		g.Eventually(func(g gomega.Gomega) {
			g.Expect(c.Get(ctx, trialKey, trial)).Should(gomega.Succeeded())
			g.Expect(trial.IsMetricsUnavailable()).Should(gomega.BeTrue)
			g.Expect(trial.Status.Observation.Metrics).ShouldNot(gomega.HaveLen(0))
			g.Expect(trial.Status.Observation.Metrics[0]).Should(gomega.BeComparableTo(commonv1beta1.Metric{
			  Name: METRCS_NAME,
			  Min: consts.UnavailableMetricValue,
			  Max: consts.UnavailableMetricValue,
			  Latest: consts.UnavailableMetricValue,
			}))
		}, timeout).Should(gomega.Succeeded())
```

Could you use the propagated gomega instance to improve debuggability?

Reply:
I guess you just copy and paste here, but we should select the ideal approach in the new codes.
If you can replace the not ideally codes with ideal forms in followup PR, it would be great.

Reply:
Yeah, I think the propagated gomega instance looks better. 
I'll raise a new issue describing it and open a PR to it later :)